// vortex-api/index.js
const fs = require('fs');
const fsPromises = require('fs').promises;

module.exports = {
  actions: {
    setModAttribute: (gameId, modId, key, value) => ({ type: 'SET_MOD_ATTRIBUTE', payload: { gameId, modId, key, value } }),
  },
  fs: {
    readdirAsync: async (p) => fsPromises.readdir(p),
    statAsync: async (p) => fsPromises.stat(p),
    readFileAsync: async (p, enc) => fsPromises.readFile(p, enc),
    writeFileAsync: async (p, data, enc) => fsPromises.writeFile(p, data, enc),
    ensureDirWritableAsync: async (p) => { try { await fsPromises.mkdir(p, { recursive: true }); return p; } catch (e) { throw e; } },
    createWriteStream: (p) => fs.createWriteStream(p),
  },
  log: function (level, ...args) {
    // Simple logger compatible with usage in the extension
    console.log(`[vortex-api][${level}]`, ...args);
  },
  util: {
    getSafe: function (obj, pathArray, def) {
      try {
        let cur = obj;
        for (const k of (pathArray || [])) {
          if (cur === undefined || cur === null) return def;
          cur = cur[k];
        }
        return (cur === undefined) ? def : cur;
      } catch (e) {
        return def;
      }
    }
  },
  types: {},
};
